arch/sim/src/sim/up_linuxi2c.h

```diff
@@ -0,0 +1,44 @@
+/****************************************************************************
+ * arch/sim/src/sim/up_linuxi2c.h
```

Comment:
we need abstract the interface between host and sim:

1. The function name called by the board files should be generic
2. The function between sim and host also need be generic
3. And different host can provide the different implementation for item 2